awscli/customizations/configure/addmodel.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import os
+
+from botocore.model import ServiceModel
+
+from awscli.customizations.commands import BasicCommand
+
+
+def _get_endpoint_prefix_to_name_mappings(session):
+    # Get the mappings of endpoint prefixes to service names from the
+    # available service models.
+    prefixes_to_services = {}
+    for service_name in session.get_available_services():
+        service_model = session.get_service_model(service_name)
```

Comment:
I am not a fan of having users know about the services whose name that deviate from the endpoint prefix because I believe that would force the user to know too much about the internals of botocore as they would have to know a service name has a different name than its endpoint prefix. Especially since with this current implementation, users do not need to even know anything about the contents of the model to get it to work. If we are to go down this route, I would much rather have ``--service-name`` be a required parameter as that would lessen the confusion of when service name needed and when its not needed.

As to the JSON file, I considered that as well, but that would add some maintenance burden to the process when we add new services. Furthermore with how I have it, the command will even identify previously added custom commands and know where to place it if this custom command has a different endpoint than what is named. As to other places where it is needed, can you point to places where this information is needed? I am having trouble thinking of any

At this point, I would prefer making ``--service-name`` required or keeping as is. If there are other locations where this endpoint mapping logic is needed, that option would become more favorable for me.